### PR TITLE
Implements component schema-editor-geojson-point using mapbox-gl-js

### PR DESCRIPTION
**Pull request by manuelroth**
This PR implements the component `schema-editor-geojson-point` using mapbox gl js in preparation for the new component `schema-editor-bbox`:
- Uses [mapbox-gl-js](https://github.com/mapbox/mapbox-gl-js) for rendering the map and [@tarekraafat/autocomplete.js](https://github.com/TarekRaafat/autoComplete.js) for the autocomplete of the search input
- Moves the map-options input to the left of the map

Screenshot 2019-07-30 at 11 15 52

### CODE REVIEW

File: client/src/elements/schema-editor/schema-editor-geojson-point.js [hunk 1]
```diff
@@ -17,29 +15,15 @@ export class SchemaEditorGeojsonPoint {
   @bindable
   showNotifications;
 
-  options = {
```
Comment (benib) [inline client/src/elements/schema-editor/schema-editor-geojson-point.js]: this option and it's handling should be kept to still support the Q-Quiz tool where this is in use to have a combination of a Point geometry and a bbox.

File: client/src/elements/schema-editor/schema-editor-geojson-point.js [hunk 2]
```diff
@@ -1,10 +1,8 @@
-import { bindable, inject, Loader, LogManager } from "aurelia-framework";
+import { bindable, inject, Loader } from "aurelia-framework";
 import QConfig from "resources/QConfig";
 import { isRequired } from "./helpers.js";
-
-const log = LogManager.getLogger("Q");
-const iconPinSvg =
-  '<svg viewBox="0 0 52 52"><path d="M31.5,35.5 M20.5,16.5 M20.5,35.5 M31.5,16.5 M30,18c0-2.2-1.8-4-4-4s-4,1.8-4,4c0,1.9,1.3,3.4,3,3.9V38h1h1V21.9C28.7,21.4,30,19.9,30,18z"/></svg>';
+import mapboxgl from "mapbox-gl";
```
[the remark this answers is not in the document]
Comment (manuelroth) [reply client/src/elements/schema-editor/schema-editor-geojson-point.js]: Fixed in https://github.com/nzzdev/Q-editor/pull/225/commits/d0afa1e4d0582d143eaec4e3d099466f2104f8f6

File: client/src/elements/schema-editor/schema-editor-geojson-point.js [hunk 3]
```diff
@@ -1,10 +1,8 @@
...
+import Autocomplete from "@tarekraafat/autocomplete.js";
```
[the remark this answers is not in the document]
Comment (manuelroth) [reply client/src/elements/schema-editor/schema-editor-geojson-point.js]: Fixed in https://github.com/nzzdev/Q-editor/pull/225/commits/d0afa1e4d0582d143eaec4e3d099466f2104f8f6

File: client/src/elements/schema-editor/schema-editor-geojson-point.js [hunk 4]
```diff
@@ -53,335 +37,179 @@ export class SchemaEditorGeojsonPoint {
...
-    this.map = window.Leaflet.map(this.mapContainer, {
-      zoomControl: false, // we add this later after the search control
-      touchZoom: "center",
-      scrollWheelZoom: "center"
-    }).setView([47.36521171867246, 8.547435700893404], 13);
-
-    // we need to keep the current search results to use the label after selection
-    let currentSearchResults = [];
-
-    if (!schemaEditorConfig.geojson.opencagedata.apiKey) {
```
[the remark this answers is not in the document]
Comment (manuelroth) [reply client/src/elements/schema-editor/schema-editor-geojson-point.js]: Fixed in https://github.com/nzzdev/Q-editor/pull/225/commits/01ab7c4d2da8657aa88269df8d79c6266df04a61

File: client/src/elements/schema-editor/schema-editor-geojson-point.js [hunk 5]
```diff
@@ -53,335 +39,251 @@ export class SchemaEditorGeojsonPoint {
...
         this.updateMarker();
-        if (this.change) {
-          this.change();
+        if (this.options.bbox === "manual") {
+          this.updateBBox();
         }
       }
     });
 
-    const layerConfig = schemaEditorConfig.geojson.layer;
```
Comment (benib) [inline client/src/elements/schema-editor/schema-editor-geojson-point.js]: If the support for a tilelayer config gets removed, I would consider this a breaking change. If it is not too crazy to support the new config with a style url and the old one with a tile layer config (mapbox-gl can still display normal raster tile layers right?) then I would prefer to keep the support for this as it makes it easier to update the editor for all instances we are running.

**Pull request merged**